projects/core/components/hosted-dropdown/hosted-dropdown.component.ts

```diff
@@ -106,6 +118,14 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
         return this.openChange.value;
     }
 
+    get isLastOpenedDropdown(): boolean {
+        return !!this.dropdown && !this.dropdown?.nextElementSibling;
```

Comment:
Unnecessary test for `!!this.dropdown` if we will test for `this.open` anyway. `dropdown` is already public, let's just move this test for `nextElementSibling` inside `shouldEscape`. `shouldClose` is also probably a better name for that predicate :)